File: kotlin/algorithms/search/JumpSearch.kt
package algorithms.search

import kotlin.math.sqrt

fun jumpSearch(input: List<Int>, expectedValue: Int): Int {
    val step = sqrt(input.size.toDouble()).toInt()
    var foundIndex = 0
    var previousIndex = 0

    while (input[foundIndex] < expectedValue) {
        previousIndex = foundIndex
        foundIndex += step
    }

    for (i in previousIndex..foundIndex) {
        if (input[i] == expectedValue) {
            foundIndex = i
            break
        }
    }

    return foundIndex
}

File: dart/algorithms/search/jump_search.dart
import 'dart:math';

int binarySearch(List<int> input, int expectedValue) {
  final step = sqrt(input.length).toInt();
  var foundIndex = 0;
  var previousIndex = 0;

  while (input[foundIndex] < expectedValue) {
    previousIndex = foundIndex;
    foundIndex += step;
  }

  for (int i = previousIndex; i < foundIndex; i++) {
    if (input[i] == expectedValue) {
      foundIndex = i;
      break;
    }
  }

  return foundIndex;
}